### PR TITLE
feat: improve test coverage

### PR DESCRIPTION
**Pull request by nshaaban-cPacket**

### CODE REVIEW

File: src/engineio/transports/polling.rs [hunk 1]
```diff
@@ -87,34 +87,55 @@ impl Transport for PollingTransport {
...
-        assert_ne!(transport.base_url()?, url);
-        /*
-        //TODO: Make sure it doesn't double up
-        transport.set_base_url("127.0.0.1/?transport=polling".to_owned())?;
-        assert_eq!(transport.base_url()?, "127.0.0.1/?transport=polling");
-        assert_ne!(transport.base_url()?, url);
-        */
+        println!("{:?}", transport);
+        let test: Box<dyn Transport> = Box::new(transport);
+        println!("{:?}", test);
```
Comment (1c3t3a) [inline src/engineio/transports/polling.rs:138]: Why you're looking for a debug test here? I don't really understand why there are the printlns

File: src/engineio/transports/polling.rs [hunk 2]
```diff
@@ -87,34 +87,55 @@ impl Transport for PollingTransport {
     }
 
     fn set_base_url(&self, base_url: Url) -> Result<()> {
-        *self.base_url.write()? = base_url;
+        let mut url = base_url;
+        if url
+            .query_pairs()
+            .filter(|(k, v)| k == "transport" && v == "polling")
+            .count()
+            == 0
```
Comment (1c3t3a) [inline src/engineio/transports/polling.rs:95]: You could use `any` here which should be faster as it returns whenever it found an occurrence of the thing you're looking at, other that `filter`+ `count` which looks at every element.

File: src/engineio/transports/websocket.rs [hunk 3]
```diff
@@ -109,31 +109,51 @@ impl Transport for WebsocketTransport {
...
+        let mut url = url;
+        if url
+            .query_pairs()
+            .filter(|(k, v)| k == "transport" && v == "websocket")
+            .count()
+            == 0
+        {
+            url.query_pairs_mut().append_pair("transport", "websocket");
+        }
+        url.set_scheme("ws").unwrap();
```
Comment (1c3t3a) [inline src/engineio/transports/websocket.rs:121]: The same as above ;)

File: src/engineio/transports/websocket_secure.rs [hunk 4]
```diff
@@ -114,27 +114,56 @@ impl Transport for WebsocketSecureTransport {
...
+        let mut url = url;
+        if url
+            .query_pairs()
+            .filter(|(k, v)| k == "transport" && v == "websocket")
+            .count()
+            == 0
+        {
+            url.query_pairs_mut().append_pair("transport", "websocket");
+        }
+        url.set_scheme("wss").unwrap();
```
Comment (1c3t3a) [inline src/engineio/transports/websocket_secure.rs:126]: Same as above ;)

**Review by 1c3t3a — CHANGES_REQUESTED** [review]

File: src/engineio/transports/polling.rs (hunk 2 above)
Comment (nshaaban-cPacket) [reply src/engineio/transports/polling.rs]: Good optimization!

File: src/engineio/transports/polling.rs (hunk 1 above)
Comment (nshaaban-cPacket) [reply src/engineio/transports/polling.rs]: Woops, forgot to implement the actual test. https://github.com/1c3t3a/rust-socketio/pull/80/commits/097ee02bd141e6cc87b14a79218efe7761cdadfc

**Review by 1c3t3a — APPROVED** [review]

**Pull request merged**